libs/common/src/components/modals/signature-modal/multisig-confirm-messages.tsx

```diff
@@ -40,6 +41,96 @@ export const MultisigConfirmMessages = observer<MultisigConfirmMessagesProps>(
 
     const [loading, setLoading] = useState(false);
 
+    const getKeysData = () =>
```

Comment:
Same here: three `props.isLogin` checks and you already know that `ConfirmMessages` also has various of those checks. So splitting here makes the code more readable in both branches.